internal/renderers/skia/itemrenderer.rs

```diff
@@ -65,31 +65,39 @@ impl<'a> SkiaItemRenderer<'a> {
         width: PhysicalLength,
         height: PhysicalLength,
     ) -> Option<skia_safe::Paint> {
-        Self::brush_to_shader(brush, width, height).map(|shader| {
-            let mut paint = skia_safe::Paint::default();
+        Self::brush_to_shader(brush, width, height, |mut paint, shader| {
             paint.set_shader(shader);
             paint.set_alpha_f(paint.alpha_f() * self.current_state.alpha);
+
             paint
         })
     }
 
-    fn brush_to_shader(
+    fn brush_to_shader<T>(
         brush: Brush,
         width: PhysicalLength,
         height: PhysicalLength,
-    ) -> Option<skia_safe::shader::Shader> {
+        func: impl FnOnce(skia_safe::Paint, Option<skia_safe::Shader>) -> T,
```

Comment:
This is a bit of a strangle pattern.
Why not having a function with this signature: `brush_to_shader(...) -> Option(Paint, Shader)`  instead of taking a callback?
Then it can be used like so:
```rust
let (mut paint, shader) = Self::brush_to_shader(...)?;
```

Anyway it seems that the Shader cannot be None when the function is called, so no need to take an `Option` as argument.

Reply:
Just personal preference, I felt like `colorize_image` was 2% clearer with the closure argument over a tuple return value, but I've been living mostly outside of rust lately and that was probably my being overly accustomed to foreign idioms. 

Adjusted to match your feedback.

Reply:
Re: taking an `Option` as an argument, `brush_to_paint()` is more frequently referenced/called in the source and operates on an `Option<shader>`, so the cruft in `colorize_ image()`, to me, was simply picking favorites between the two callers. Happy to knock a `?` from one and add a `Some` to the other if you find that preferable.